Use #pragma once and update to Clang-format 10

Flamefire opened the pull request:
Closes #1302

Flow86 commented:
```
fatal error: in "SystemTestSuite/GetUsername": critical check !System::getUserName().empty() has failed
```

Flow86 commented on .github/workflows/static-analysis.yml:
```diff
@@ -7,39 +7,39 @@ on:
+      - name: Install dependencies
+        run: |
+          sudo apt-get update && sudo apt-get install -y \
+            clang-9 clang-tidy-9 \
```
shouldn't it install clang 10?

Flamefire replied:
Not yet. clang and clang-tidy must match and clang-tidy-10 contains checks that I want to enable in the near future but which would fail now. Note that clang-format-10 is not dependent on clang-10

Flamefire commented:
@Flow86 Why would `$USER` be empty on jenkins? Even `getpwuid` doesn't seem to work.

Flow86 commented:
Hmm. docker environment? I don't know. I will check

Flamefire commented:
Yep, looks very much like it. I think we should even throw an exception if the user name is empty, similar to Windows. When we use $USER then for docker you can just pass something to the docker env as $USER. Is likely the easiest. Then I'd revert my commit  using getpwuid

Flamefire commented:
Added a workaround in #1303: Set $USER manually if not set. Should be ok, will remove the commit to libutil using the syscall

Flamefire commented:
Extracted the jenkins fix to https://github.com/Return-To-The-Roots/s25client/pull/1305, I can rebase the other 2 PRs onto that.

For this one I updated clang-tidy to version 10 too to resolve the version disparity.

@Flow86 Please review and merge as appropriate
